pkg/didcomm/messenger/messenger.go

```diff
@@ -138,6 +138,8 @@ func (m *Messenger) Send(msg service.DIDCommMsgMap, myDID, theirDID string) erro
 		return fmt.Errorf("save metadata: %w", err)
 	}
 
+	delete(msg, jsonThread)
```

Comment:
Why?

Reply:
The thread should not be in the message. The messenger is responsible for that. The function Send starts a new thread. Which means that the message cannot contain a custom thread.

Reply:
@soluchok sorry - I'm not following here. I don't see where a `~thread` decorator will be included in the message finally sent over the wire.

Reply:
I also am not seeing thread handling in the codepath.